opentracing-usecases/src/main/java/io/opentracing/usecases/AutoFinishScopeManager.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.usecases;
+
+import io.opentracing.ContinuableScope;
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import io.opentracing.Span;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Created by bhs on 8/5/17.
+ */
+public class AutoFinishScopeManager implements ScopeManager {
```

Comment:
Maybe I am misunderstanding something, but this seems to be violating the stated goal in #166  of not forcing refcounting on instrumentation. Even though this class is optional, once it's set as a scope manager on the tracer (a decision made by the application) it becomes mandatory for all instrumentations (which may not be written by the application author).

I thought we were going for a pattern where refcounting can be explicitly requested, e.g. by wrapping the Scope

Reply:
It seems to me this scope manager was included in a `usecases` package so it wouldn't be really included in the final merge, but still used as a test case. Wondering if the `ThreadLocalScopeManager` (which is *actually* included in `opentracing-util`) would be fine for you?